typed/typed.go

```diff
@@ -187,7 +201,41 @@ func (tv TypedValue) RemoveItems(items *fieldpath.Set) *TypedValue {
 }
 
 // ExtractItems returns a value with only the provided list or map items extracted from the value.
-func (tv TypedValue) ExtractItems(items *fieldpath.Set) *TypedValue {
+func (tv TypedValue) ExtractItems(items *fieldpath.Set, opts ...extractItemsOption) *TypedValue {
+	options := &extractItemsOptions{}
+	for _, opt := range opts {
+		opt(options)
+	}
+	if options.appendKeyFields {
+		tvPathSet, err := tv.ToFieldSet()
+		if err != nil {
+			// It is safe to continue and skip appending the key fields, because
+			// if there is error to convert to fieldPath.Set, there is
+			// no way to find the key fields values by following the schema.
+			goto startExtrcation
```

Comment:
```suggestion
			goto startExtraction
```

Reply:
IMO, it would be preferable to return the error and avoid the `goto` than failing silently. I understand that it would change the signature and break consumers. Maybe a new function?

Reply:
It is a good practice to expose the error. But I'm not a fan to add a new interface here. We should address the non-error return interface(`RemoveItems`,`ExtractItems`) in another PR. wdyt?

Reply:
LGTM. 👍🏻